src/cmd-buildextend-gcp

```diff
@@ -7,94 +7,140 @@ extending the meta.json with GCP image name.
+        tmp_img_gcp_raw = os.path.join(TMPDIR, 'disk.raw')
+        tmp_img_gcp_tar = os.path.join(TMPDIR, self.gcp_tarball_name)
+        run_verbose([f"{cosa_dir}/gf-platformid",
+                     img_qemu, tmp_img_gcp, plat_id])
+        run_verbose(['qemu-img', 'convert', '-f', 'qcow2', '-O', 'raw',
+                    tmp_img_gcp, tmp_img_gcp_raw])
+        run_verbose(['tar', '-C', f"{TMPDIR}", '-Sczf', f"{tmp_img_gcp_tar}",
+                    f"disk.raw"])
+        os.unlink(tmp_img_gcp)
+        os.unlink(tmp_img_gcp_raw)
+
+        fsize = subprocess.check_output([
+            "stat", "--format", '%s', tmp_img_gcp_tar])
+        log.debug(" * calculating checksum")
+        self._found_files[tmp_img_gcp_tar] = {
+                "local_path": os.path.abspath(tmp_img_gcp_tar),
+                "path": os.path.basename(tmp_img_gcp_tar),
+                "md5": md5sum_file(tmp_img_gcp_tar),
+                "size": int(fsize)
+        }
+        log.debug(" * size is %s", self._found_files[tmp_img_gcp_tar]["size"])
+        log.debug(" * md5 is %s", self._found_files[tmp_img_gcp_tar]["md5"])
+
+
+def cli():
+    """
+    Parse args and dispatch
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--build", help="Build ID", required=True)
+    parser.add_argument("--bucket", help="Storage account to write image to",
+                        default=os.environ.get("GCP_BUCKET"))
+    parser.add_argument(
+        "--gce", help="Use GCE as the platform ID instead of GCP",
+        action="store_true",
+        default=bool(os.environ.get("GCP_GCE_PLATFORM_ID", False)))
+    parser.add_argument("--force", help="Replace existing images and upload",
+                        action="store_true",
+                        default=bool(os.environ.get("GCP_FORCE", False)))
+    parser.add_argument("--json-key", help="GCP Service Account JSON Auth",
+                        default=os.environ.get("GCP_JSON_AUTH"))
+    parser.add_argument("--name-suffix", help="Append suffix to name",
+                        required=False)
+    parser.add_argument("--project", help="GCP Project name",
+                        default=os.environ.get("GCP_PROJECT_NAME"))
+    args = parser.parse_args()
+
+    # Argument checks for environment strings that are required
+    arg_exp_str = "parameter '--{}' or envVar '{}' must be defined"
+    if args.bucket is None:
+        raise Exception(arg_exp_str.format("bucket", "GCP_BUCKET"))
+    if args.json_key is None:
+        raise Exception(arg_exp_str.format("json-key", "GCP_JSON_AUTH"))
+    if args.project is None:
+        raise Exception(arg_exp_str.format("project", "GCP_PROJECT"))
+
+    # Identify the builds
+    build = Build(os.path.join('builds', args.build), args.build)
+
+    if 'gcp' in build.meta['images'] and not args.force:
+        print("GCP image already built; use --force to rebuild")
+        sys.exit(0)
+
+    build.build_artifacts()
```

Comment:
Should `args` be passed to this function?

Reply:
agree

Reply:
It should :+1: